test/functional/feature_dbcrash.py

```diff
@@ -256,7 +256,11 @@ def run_test(self):
             self.log.debug("Mining longer tip")
             block_hashes = []
             while current_height + 1 > self.nodes[3].getblockcount():
-                block_hashes.extend(self.nodes[3].generate(min(10, current_height + 1 - self.nodes[3].getblockcount())))
+                block_hashes.extend(self.nodes[3].generatetoaddress(
```

Comment:
How about doing the generate call on its own line? newlines are free!

Reply:
That would require poisoning the namespace with a symbol name. Named arguments and functional programming ftw. Bracket are free! :rocket:

Reply:
Really? You think 4-nested brackets are easier to read or are you just trolling?

Reply:
Not trolling. 4 brackets is how I write code. :grimacing: 

https://github.com/bitcoin/bitcoin/blame/fa2ca0cbdde5c6c5e407ec037e52e3f6315a0b37/test/functional/mempool_accept.py#L162-L168

Reply:
@MarcoFalke  lol I see a couple of places with style like this, no wonder 😆

Reply:
@glowang I'd definitely support any PRs to make them more readable :)

Reply:
Huh, is this generally considered to be less readable? If yes, I might change my style, but comparing the two:

```py
in_1 = 'foo'
in_2 = 'bar'
res = run(in_1, in_2)
```

vs

```py
res = run(
  in_1='foo',
  in_2='bar',
)
```

The latter one has a lower mental load because there is two less symbols in the surrounding scope. I can't find references, but I am pretty sure we had a bug in the tests because symbols were leaked from one test to the next test by accident.

Also, inlining promotes use of named arguments, which obviously promotes code readability. I am sure we had at least one bug due to not using named arguments.

Reply:
This isn't an honest comparison. Nesting functions 4 deep inside is very different from the example you've given. It's also completely unrelated to whether functions are called with named or positional arguments.

Reply:
The indentation @MarcoFalke  uses does help a lot, and I can see your point on reducing mental load now.

Reply:
I agree in the `in_1`, `in_2` example the second looks more clear, but I think thats mainly because we're using mocked out variable names. 

if it was something like
```
blockhash = getbestblockhash()
coinbase_txid = getblock(blockhash)['tx'][0]
res = calculate_fee_amount(coinbase_txid, blockhash) 
```
that seems pretty clear to me. The variable naming is relevant for capturing intent. In this case you could just look at the `res=calculate..` line and get an idea of what's happening (because at first pass, you can assume the variables are named correctly.) 

--

in regards to this code snippet, I'd find something like: 
```
num_blocks_to_create = min(10, current_height + 1 - self.nodes[3].getblockcount())
new_block_hashes = self.nodes[3].generatetoaddress(
  nblocks = num_blocks_to_create,
  # new address to avoid mining a block that has just been invalidated
  address=self.nodes[3].getnewaddress(),
)
block_hashes.extend(new_block_hashes)
```

significantly more clear than the current;
```
block_hashes.extend(self.nodes[3].generatetoaddress(
  nblocks=min(10, current_height + 1 - self.nodes[3].getblockcount()),
  # new address to avoid mining a block that has just been invalidated
  address=self.nodes[3].getnewaddress(),
))
```

if I were proposing the code, I'd think about the variable names a bit more... but the information they can provide about the content can be _super_ useful, esp in untyped languages. I actually looked up the docs for `generatetoaddress` to clarify what was happening here to write up the alternative. 

Coming from rubyland, this is pretty common practice for making code easier to parse. A quick google of "self documenting code" landed me on [this page](https://multi-programming.com/blog/self-documenting-code), where #1 and #2 are about extracting logic to abstract it behind a variable name (either via code or variable).

--

Lol, lots of words. I want to share because when I come across the deeply nested functions in the functional tests, I find it difficult to parse and will often re-write them into more incremental chunks just to read what's happening.